app/models/vote.rb

```diff
@@ -29,8 +29,12 @@ def self.noes
     where("votes.vote = 'no'")
   end
 
-  def rebellion?
-    !free? && vote != whip_guess
+  def rebellion?(whip = nil)
+    if whip.nil?
+      !free? && vote != whip_guess
+    else
+      !whip.free? && vote != whip.whip_guess
+    end
```

Comment:
Hum, I will try that. But not sure if it will work. `memoize` works if you call the same method repeatedly, but I am not sure if that's the case here (I don't remember I opened this a few days ago :smiley_cat: ). IIRC the `whip` was loaded first and then `vote` was loaded later. That's why I did what I did. But as I said, I will try your suggestoin regardless :smile: